control/controlclient/direct.go

```diff
@@ -208,7 +208,12 @@ func NewDirect(opts Options) (*Direct, error) {
 	if opts.Hostinfo == nil {
 		c.SetHostinfo(hostinfo.New())
 	} else {
+		ni := opts.Hostinfo.NetInfo
+		opts.Hostinfo.NetInfo = nil
```

Comment:
is this ours to mutate? should it be cloned first?

Reply:
`// non-nil passes ownership, nil means to use default using os.Hostname, etc`